src/core/a-entity.js

```diff
@@ -686,28 +688,43 @@ var proto = Object.create(ANode.prototype, {
       var arg1Type = typeof arg1;
       var clobber;
       var componentName;
+      var delimiterIndex;
       var isDebugMode;
+      var self = this;
+      var setArguments = arguments;
 
-      var pos = attrName.indexOf(MULTIPLE_COMPONENT_DELIMITER);
-      componentName = pos > 0 ? attrName.substring(0, pos) : attrName;
+      delimiterIndex = attrName.indexOf(MULTIPLE_COMPONENT_DELIMITER);
+      componentName = delimiterIndex > 0 ? attrName.substring(0, delimiterIndex) : attrName;
 
-      // Determine which type of setAttribute to call based on the types of the arguments.
-      if (COMPONENTS[componentName]) {
-        if (arg1Type === 'string' && typeof arg2 !== 'undefined') {
-          singlePropertyUpdate(this, attrName, arg1, arg2);
-        } else {
-          clobber = arg1Type !== 'object' || (arg1Type === 'object' && arg2 === true);
-          this.updateComponent(attrName, arg1, clobber);
-        }
+      // Not a component.
+      if (!COMPONENTS[componentName]) {
+        normalSetAttribute(this, attrName, arg1);
+        return;
+      }
 
-        // In debug mode, write component data up to the DOM.
-        isDebugMode = this.sceneEl && this.sceneEl.getAttribute('debug');
-        if (isDebugMode) { this.components[attrName].flushToDOM(); }
+      // Wait for compoent to be initialized with DOM data.
+      if (!this.components[attrName] && this.hasAttribute(attrName)) {
+        this.addEventListener('componentinitialized', function waitForInit (evt) {
```

Comment:
Might work. I can try. `dependencies` doesn't matter since `initComponent` will set them up. In the case another component depends on it, it's fine to `init` it a little early.